br/pkg/backup/push.go

```diff
@@ -4,213 +4,183 @@ package backup
 	"github.com/pingcap/tidb/br/pkg/utils"
-	"github.com/pingcap/tidb/pkg/util/redact"
 	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-// pushDown wraps a backup task.
-type pushDown struct {
-	mgr    ClientMgr
-	respCh chan responseAndStore
-	errCh  chan error
+type ResponseAndStore struct {
+	Resp    *backuppb.BackupResponse
+	StoreID uint64
 }
 
-type responseAndStore struct {
-	Resp  *backuppb.BackupResponse
-	Store *metapb.Store
-}
-
-func (r responseAndStore) GetResponse() *backuppb.BackupResponse {
+func (r ResponseAndStore) GetResponse() *backuppb.BackupResponse {
 	return r.Resp
 }
 
-func (r responseAndStore) GetStore() *metapb.Store {
-	return r.Store
+func (r ResponseAndStore) GetStoreID() uint64 {
+	return r.StoreID
 }
 
-// newPushDown creates a push down backup.
-func newPushDown(mgr ClientMgr, capacity int) *pushDown {
-	return &pushDown{
-		mgr:    mgr,
-		respCh: make(chan responseAndStore, capacity),
-		errCh:  make(chan error, capacity),
-	}
-}
-
-// FullBackup make a full backup of a tikv cluster.
-func (push *pushDown) pushBackup(
+func doSendBackup(
 	ctx context.Context,
+	client backuppb.BackupClient,
 	req backuppb.BackupRequest,
-	prTree rtree.ProgressRangeTree,
-	stores []*metapb.Store,
-	checkpointRunner *checkpoint.CheckpointRunner[checkpoint.BackupKeyType, checkpoint.BackupValueType],
-	progressCallBack func(),
+	respFn func(*backuppb.BackupResponse) error,
 ) error {
-	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
-		span1 := span.Tracer().StartSpan("pushDown.pushBackup", opentracing.ChildOf(span.Context()))
-		defer span1.Finish()
-		ctx = opentracing.ContextWithSpan(ctx, span1)
-	}
-
-	// Push down backup tasks to all tikv instances.
-	failpoint.Inject("noop-backup", func(_ failpoint.Value) {
-		logutil.CL(ctx).Warn("skipping normal backup, jump to fine-grained backup, meow :3", logutil.Key("start-key", req.StartKey), logutil.Key("end-key", req.EndKey))
-		failpoint.Return(nil)
+	failpoint.Inject("hint-backup-start", func(v failpoint.Value) {
+		logutil.CL(ctx).Info("failpoint hint-backup-start injected, " +
+			"process will notify the shell.")
+		if sigFile, ok := v.(string); ok {
+			file, err := os.Create(sigFile)
+			if err != nil {
+				log.Warn("failed to create file for notifying, skipping notify", zap.Error(err))
+			}
+			if file != nil {
+				file.Close()
+			}
+		}
+		time.Sleep(3 * time.Second)
 	})
-
-	wg := new(sync.WaitGroup)
-	errContext := utils.NewErrorContext("pushBackup", 10)
-	for _, s := range stores {
-		store := s
-		storeID := s.GetId()
-		lctx := logutil.ContextWithField(ctx, zap.Uint64("store-id", storeID))
-		if err := utils.CheckStoreLiveness(s); err != nil {
-			logutil.CL(lctx).Warn("skip store", logutil.ShortError(err))
-			continue
+	bCli, err := client.Backup(ctx, &req)
+	failpoint.Inject("reset-retryable-error", func(val failpoint.Value) {
+		switch val.(string) {
+		case "Unavaiable":
```

Comment:
Shuold be "Unavailable"?

Reply:
seems this failpoint never works before..